### DISCUSSION TITLE
Generate forms directly in the Publicodes Studio

### DISCUSSION THREAD

**Discussion by mquandalle** [publicodes/publicodes#402] [General]
One interesting use case of publicodes is to generate interactive forms. @revolunet has created great examples of that in https://socialgouv.github.io/demo-publicodes. Betagouv teams also use publicodes to generate these form: https://mon-entreprise.urssaf.fr/g%C3%A9rer/demande-mobilit%C3%A9 and https://code.travail.gouv.fr/outils/preavis-demission

It would be great to be able to generate these kind of forms directly in the publicodes studio. To implement this feature we need to:  
- standardize how we define meta-data on rules to customize the input fields
- have a way to see and edit the `situation` object in the studio

**Comment by revolunet** [comment]
react-jsonschema-form may have an interesting approach with an optional [uiSchema](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/uiSchema/) definition key that hold the form customisation

**Comment by laem** [comment]
> standardize how we define meta-data on rules to customize the input fields

I'm not sure it's needed for a first basic implementation :)
